silkrpc/core/gas_price_oracle_test.cpp

```diff
@@ -0,0 +1,593 @@
+
+    block_with_hash.block.transactions.resize(2);
+    block_with_hash.block.transactions[0].max_priority_fee_per_gas = max_priority_fee_per_gas_tx1;
+    block_with_hash.block.transactions[0].max_fee_per_gas = max_fee_per_gas_tx1;
+    block_with_hash.block.transactions[0].from = kFromTnx1;
+
+    block_with_hash.block.transactions[1].max_priority_fee_per_gas = max_priority_fee_per_gas_tx2;
+    block_with_hash.block.transactions[1].max_fee_per_gas = max_fee_per_gas_tx2;
+    block_with_hash.block.transactions[1].from = kFromTnx2;
+
+    return block_with_hash;
+}
+
+static void clear_blocks_vector() {
+    kBlocks.clear();
+}
+
+static void fill_blocks_vector(uint16_t size,
+    const evmc::address &beneficiary,
+    const intx::uint256 &base_fee,
+    const intx::uint256 &max_priority_fee_per_gas_tx1,
+    const intx::uint256 &max_fee_per_gas_tx1,
+    const intx::uint256 &max_priority_fee_per_gas_tx2,
+    const intx::uint256 &max_fee_per_gas_tx2) {
+
+    kBlocks.reserve(size);
+    for (auto idx = 0; idx < size; idx++) {
+        silkworm::BlockWithHash block_with_hash = allocate_block(idx, beneficiary, base_fee,
+            max_priority_fee_per_gas_tx1, max_fee_per_gas_tx1,
+            max_priority_fee_per_gas_tx2, max_fee_per_gas_tx2);
+        kBlocks.push_back(block_with_hash);
+    }
+}
+
+static void fill_blocks_vector(uint16_t size,
+    const evmc::address &beneficiary,
+    const intx::uint256 &base_fee,
+    const intx::uint256 &max_priority_fee_per_gas,
+    const int delta_max_priority_fee_per_gas,
+    const intx::uint256 &max_fee_per_gas,
+    const int delta_max_fee_per_gas) {
+
+    kBlocks.reserve(size);
+    for (auto idx = 0; idx < size; idx++) {
+        int64_t max_priority = int64_t(max_priority_fee_per_gas) + delta_max_priority_fee_per_gas * idx;
+        if (max_priority < 0) {
+            max_priority = 0;
+        }
+        int64_t  max_fee = int64_t(max_fee_per_gas) + delta_max_fee_per_gas * idx;
+        if (max_fee < 0) {
+            max_fee = 0;
+        }
+
+        silkworm::BlockWithHash block_with_hash = allocate_block(idx, beneficiary, base_fee,
+            intx::uint256{max_priority},
+            intx::uint256{max_fee},
+            intx::uint256{max_priority},
+            intx::uint256{max_fee});
+        kBlocks.push_back(block_with_hash);
+    }
+}
+
+asio::awaitable<silkworm::BlockWithHash> get_block(uint64_t block_number) {
+    REQUIRE(kBlocks.size() > block_number);
+
+    const silkworm::BlockWithHash block = kBlocks[block_number];
+
+    co_return block;
+}
+
+BlockProvider block_provider = [](uint64_t block_number) {
+    return get_block(block_number);
+};
+
+asio::thread_pool pool{1};
+
+using Catch::Matchers::Message;
+TEST_CASE("0 blocks") {
+    const intx::uint256 base_fee = 0;
+    const intx::uint256 max_priority_fee_per_gas = 0x32;
+    const intx::uint256 max_fee_per_gas = 0x32;
+    const intx::uint256 expected_price = kDefaultPrice;
```

Comment:
Probably defining a module-level `struct` to agglomerate these parameters could be useful to improve readability

Reply:
Good suggestion